### PR TITLE
Psp 5708 generate h120 v1

### PR DESCRIPTION
**Pull request by devinleighsmith**
requires

### CODE REVIEW

**Comment by devinleighsmith** [comment]
Note that this PR is dependent on psp-4288

File: source/backend/api/Areas/DocumentGeneration/DocumentGenerationRequest.cs [hunk 1]
```diff
@@ -17,5 +17,10 @@ public class DocumentGenerationRequest
         /// get/set - The template type to use for generation.
         /// </summary>
         public FormDocumentType TemplateType { get; set; }
+
+        /// <summary>
+        /// get/set - the type the document template should be converted to when returned from cdogs.
+        /// </summary>
+        public ConvertToTypes? ConvertToType { get; set; }
```
Comment (devinleighsmith) [inline source/backend/api/Areas/DocumentGeneration/DocumentGenerationRequest.cs:24]: allows using an easy to edit word template to generate pdfs.

File: source/backend/tests/unit/api/Controllers/H120Category/H120CategoryControllerTest.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        public H120CategoryControllerTest()
+        {
+            _helper = new TestHelper();
+            _controller = _helper.CreateController<H120CategoryController>(Permissions.CompensationRequisitionView);
+            _service = _helper.GetService<Mock<IH120CategoryService>>();
+            _mapper = _helper.GetService<IMapper>();
+        }
+
+        [Fact]
+        public void GetTakesByPropertyId_Success()
```
Comment (asanchezr) [inline source/backend/tests/unit/api/Controllers/H120Category/H120CategoryControllerTest.cs:42]: nit: please rename to something relevant - eg "GetH120Categories"

File: source/frontend/src/mocks/mockCompReqH120s.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { Api_CompensationFinancial } from 'models/api/Compensation';
+
+export const mockCompReqH120s: () => Api_CompensationFinancial[] = () => [
```
Comment (asanchezr) [inline source/frontend/src/mocks/mockCompReqH120s.ts:3]: nit: please rename file to follow convention to have "*.mock" on filename

File: source/frontend/src/mocks/mockH120Categories.ts [hunk 4]
```diff
@@ -0,0 +1,360 @@
+export const getMockH120Categories = () => [
```
Comment (asanchezr) [inline source/frontend/src/mocks/mockH120Categories.ts:1]: same comment re: mock filename

**Review by asanchezr — APPROVED** [review]

**Pull request merged**